oak_sev_guest/src/guest.rs

```diff
@@ -244,6 +244,191 @@ pub enum MessageType {
     TccInfoReqsponse = 18,
 }
 
+/// Request for a derived key.
+///
+/// See Table 17 in <https://www.amd.com/system/files/TechDocs/56860.pdf>.
+#[repr(C)]
+#[derive(Debug, AsBytes, FromBytes)]
+pub struct KeyRequest {
+    /// Selects which key will be used to derive the key.
+    ///
+    /// This contains the ROOT_KEY_SELECT and KEY_SEL bit-fields. To interact with the individual
+    /// bit-fields use `KeyRequest::get_key_select`, `KeyRequest::get_root_key_select`,
+    /// `KeyRequest::set_key_select` or `KeyRequest::set_root_key_select`
+    key_select: u32,
+    /// Reserved, must be zero.
+    _reserved: u32,
+    /// Mask indicating which guest data will be mixed into the derived key.
+    ///
+    /// Use `KeyRequest::get_guest_field_select_flags` to try to convert this to a
+    /// `GuestFieldFlags` enum.
+    pub guest_field_select: u64,
+    /// The VM Protection Level (VMPL) to mix into the derived key.
+    ///
+    /// Must be greater or equal to the current VMPL and at most 3.
+    pub vmpl: u32,
+    /// The guest security version number (SVN) to mix into the key.
+    ///
+    /// Must not exceed the guest SVN provided at launch in the ID block.
+    pub guest_svn: u32,
+    /// The TCB version to mix into the key.
+    ///
+    /// Must not exceed the committed TCB.
+    pub tcb_version: u64,
+}
+
+static_assertions::assert_eq_size!(KeyRequest, [u8; 32]);
+
+/// The bit mask for the root key select bit.
+const ROOT_KEY_SELECT_MASK: u32 = 1 << 0;
```

Comment:
Maybe move these into the `impl` for namespacing.

Reply:
Done.